cpu/stm32/periph/i2c_2.c

```diff
@@ -119,8 +119,8 @@ static void _init_pins(i2c_t dev)
         i2c_config[dev].sda_pin == GPIO_PIN(PORT_B, 9)) {
         /* The remapping periph clock must first be enabled */
         RCC->APB2ENR |= RCC_APB2ENR_AFIOEN;
-        /* Then the remap can occur */
-        AFIO->MAPR |= AFIO_MAPR_I2C1_REMAP;
+        /* Then the remap can occur with SWJ_CFG in reset state */
+        AFIO->MAPR = (AFIO->MAPR & 0x001FFFFF) | AFIO_MAPR_I2C1_REMAP;
```

Comment:
This needs some explaining. The code before would read the value (including the fields of the AFIO register that are reserved for future use), set the `AFIO_MAPR_I2C1_REMAP` bit, and write the result back.

If the reserved for future use bits would get a meaning in later STM32F1 versions (which I doubt will ever be released, ST seems more focused on newer families), the new code would wipe whatever setting there has been. So, this would be a regression compared to the current upstream code.

If this fixes any issue for you, it likely is that the change in timing solves the issue. You could try adding a few `nop`s instead of this change to see if this really is why it fixes an issue for you. If so, the bug you are facing is something else.

Btw.: It would be helpful if you could add some details on what the actual issue is.

There is indeed something possible bug in the AFIO access: The read-modify-write should be done atomic here. Otherwise concurrent updates to the AFIO register may get lost. (E.g. if the remapping for I2C pins is set while at the same time the SPI pins is done, one update may cause the other to be lost.) That said, typically peripherals are initialized during boot up during `periph_init()` - so way before a second thread is actually started. Only applications manually calling `i2c_init()` would have trigger this issue.

Reply:
Yes sorry for the lack of explanation, I realize now that the details I had added about the changes were commented out.  Thanks for your patience.

Reply:
No problem. I should also have checked the issue tracker first :)